### PR TITLE
github: add Github Actions workflow for tests; support in vet.sh

### PR DESCRIPTION
**Pull request by dfawley**

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.14
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      # Run the vet checks.
+      - name: vet
+        run: ./vet.sh -install && ./vet.sh
```
Comment (menghanl) [inline .github/workflows/testing.yml:32]: This workflow runs vet (gofmt/golint...) twice now.
Should we move the proto regen part out to it's own job?

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          - type: retry
+            goversion: 1.14
+          - type: extras
+            goversion: 1.14
+          - type: tests
+            goversion: 1.13
+          - type: tests
+            goversion: 1.12
+          - type: tests
+            goversion: 1.11  # Keep until interop tests no longer require Go1.11
```
Comment (menghanl) [inline .github/workflows/testing.yml:58]: Travis actions, lol

**Review by menghanl — APPROVED** [review]

File: .github/workflows/testing.yml (hunk 1 above)
Comment (dfawley) [reply .github/workflows/testing.yml:32]: I was thinking as a follow-up change I would add an option to vet to make it ONLY run the proto checks.  WDYT?

Comment (menghanl) [reply .github/workflows/testing.yml:32]: SG.

Do the proto checks need anything from vet.sh? Or just move it to a separate `.sh`

Comment (dfawley) [reply .github/workflows/testing.yml:32]: They both need the checks up front to make sure your branch is clean & revert any changes it makes to it.

I would probably keep them together to support the "running offline" use-case better.  `VET_ONLY_PROTO=1` or something?

**Pull request merged**